src/node.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include<iostream>
+
+template<class ValueType>
+class Node
+{
+private:
+    ValueType value;
+
+public:
+    Node* parent;
+    Node* left;
+    Node* right;
+
+    Node(ValueType v);
+    Node(ValueType v, Node<ValueType>* p);
+    ~Node() = default;
+
+    ValueType getValue();
+    void setValue(ValueType v);
+
+    friend std::ostream& operator<<(std::ostream& str, Node<ValueType>& n) {
+        return (str << n.value);
+    };
+    friend std::ostream& operator<<(std::ostream& str, Node<ValueType>* n) {
+        return (str << n->value);
+    };
+};
+
+// Constructor with value
```

Comment:
Rozumiem, że ten komenty to tylko na teraz tak orientacyjnie, ale potem jak będziemy oddawać kod to je usuniemy, nie?

Reply:
No raczej tak